pandas/io/formats/style_render.py

```diff
@@ -450,6 +461,105 @@ def _generate_col_header_row(self, iter: tuple, max_cols: int, col_lengths: dict
 
         return index_blanks + column_name + column_headers
 
+    def _generate_descriptor_row(self, iter: tuple, max_cols: int):
+        """
+        Generate the row containing calculated descriptor values for columns:
+
+         +----------------------------+---------------+---------------------------+
+         |  index_blanks ...          | descriptor_i  |  value_i by col           |
+         +----------------------------+---------------+---------------------------+
+
+        Parameters
+        ----------
+        iter : tuple
+            Looping variables from outer scope
+        max_cols : int
+            Permissible number of columns
+
+        Returns
+        -------
+        list of elements
+        """
+
+        r, descriptor = iter
+
+        # number of index blanks is governed by number of hidden index levels
+        index_blanks = [
+            _element("th", self.css["blank"], self.css["blank_value"], True)
+        ] * (self.index.nlevels - sum(self.hide_index_) - 1)
+
+        if isinstance(descriptor, str):
+            name: str | None = descriptor
+            func: Callable = getattr(Series, descriptor)
+        elif isinstance(descriptor, tuple):
+            name, func = descriptor[0], descriptor[1]
+        else:
+            name, func = None, descriptor
+
+        precision = get_option("styler.format.precision")
+        thousands = get_option("styler.format.thousands")
+        display_func: Callable = lambda x: _default_formatter(
+            func(x), precision=precision, thousands=thousands
+        )
+
+        base_css = f"{self.css['descriptor_name']} {self.css['descriptor']}{r}"
+        descriptor_name = [
+            _element(
+                "th",
+                base_css
+                if (name is not None and not self.hide_column_names)
+                else f"{self.css['blank']} {base_css}",
+                name
+                if (name is not None and not self.hide_column_names)
+                else self.css["blank_value"],
+                not all(self.hide_index_),
+            )
+        ]
+
+        descriptor_values, visible_col_count = [], 0
+        for c, col in enumerate(self.columns):
+            if c not in self.hidden_columns:
+                header_element_visible = True
+                visible_col_count += 1
+                try:
+                    header_element_value = display_func(self.data[col])
+                except Exception:
```

Comment:
Why not be strict here and raise?

Reply:
in the interest of a simpler initial PR i did not include a subset argument so the function needs to work on all colums. some dtypes might bot be suitable for the function so best approach is silent raise i think. descriptors aim to be more helpful than precise, so i think raising when failing would be annoying

Reply:
I see. What you think about raising a `UserWarning` that applying `func` failed for a column? Would that be too noisy?

Reply:
I still think so, but even if it wasn't, the way to turn it off is difficult:
  - either a pandas global option is created: I think that's a bit messy for such a small thing.
  - or since the descriptors are calc'ed at render time a `show_warnings` kwarg would have to be saved in `Styler` namespace and evaluated upon render: still a bit messy.

However, with you prompting me to think about it, probably means better documentation and better examples could highlight this and make the behaviour (and workarounds) clear.